person-api/jobs.py

```diff
@@ -0,0 +1,65 @@
+"""Populate data for connection method table periodically."""
+
+# XXX TBD deprecate this by redesigning the vault schema.
+# XXX TBD develop process for backup and restore of current vault.
+
+import boto3
+import logging
+import public
+import vault
+
+logger = logging.getLogger(__name__)
+
+
+def _get_connection_method(user):
+    if user.get('primaryEmail').split('@')[1] == 'mozilla.com':
+        return 'ad'
+    elif 'mozilliansorg_nda' in user.get('groups'):
+        return 'github'
+    else:
+        return user.get('user_id').split('|')[0]
+
+def _generate_public_struct(user):
+    public_user_data = {
+        'user_email': user.get('primaryEmail'),
+        'connection_method': _get_connection_method(user)
+    }
+
+    return public_user_data
+
+def _score_security(connection_method):
```

Comment:
we need a centralized logic for this (i propose personapi to be that) so that we dont replicate code and get different logic/pathways.
in this case, if you authenticate to google (for ex go to gmail and authenticate with LDAP) it will pop up the auth0 auth, auth you with LDAP and return that to google, albeit effectively your user_id is a google user_id, not LDAP.

If we forbid that to happen and force you to go to LDAP, you can no longer login to gmail, which.. is a problem. So instead, we detect it in auth0, and let you login with Google because we verified its a Gogle account backed by LDAP.
Here is the logic for that: https://github.com/mozilla-iam/auth0-deploy/blob/master/rules/force-users-login-most-secure-method.js#L36-L42

Note: one piece is missing in this logic => group data is NOT passed over. This works fine for Google apps because these source groups in the background with GAM (or whatever the name of their connector software now is) but this doesn't work with other apps, say, Discourse.

For that, I though of basically filling in group data back to auth0 with the CIS auth0 connector, and doing a write-back to the vault directly as well through an Auth0 CIS publisher eventually.

Reply:
Another note, regarding the Auth0 rule logic I linked: i'd basically call person-api instead so that we have a central piece of code governing that logic